src/Compilers/CSharp/Portable/Emitter/Model/SynthesizedPrivateImplementationDetailsStaticConstructor.cs

```diff
@@ -51,22 +52,61 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
                 body.Add(payloadInitialization);
             }
 
-            // MVID = TypeOf(PrivateImplementationDetails).Module.ModuleVersionId;
-
-            BoundStatement mvidInitialization =
-                factory.Assignment(
-                    factory.ModuleVersionId(),
-                    factory.Property(
-                        factory.Property(
-                            factory.TypeOfPrivateImplementationDetails(),
-                            WellKnownMember.System_Type__Module),
-                        WellKnownMember.System_Reflection_Module__ModuleVersionId));
-            body.Add(mvidInitialization);
+            // Initialize the module version ID (MVID) field. Dynamic instrumentation requires the MVID of the executing module, and this field makes that accessible.
+            body.Add(InitializeMVID(factory, diagnostics));
 
             BoundStatement returnStatement = factory.Return();
             body.Add(returnStatement);
 
             factory.CloseMethod(factory.Block(body.ToImmutableAndFree()));
         }
+
+        private BoundStatement InitializeMVID(SyntheticBoundNodeFactory factory, DiagnosticBag diagnostics)
```

Comment:
Indeed. Apparently the diagnostics parameter became unnecessary after introducing a helper for StaticCall, and I failed to notice this.